### PR TITLE
Confirm that a discount has a corresponding voucher for enterprise scenarios

### PR DESCRIPTION
**Pull request by mattdrayer**
@clintonb @mjfrey -- as discussed in https://github.com/edx/ecommerce/pull/1127#discussion_r109728419 I've added a check for a voucher object on the discount.  Let me know if this will address the short term need and we'll continue the work on https://github.com/edx/ecommerce/pull/1218.

### CODE REVIEW

File: ecommerce/extensions/fulfillment/modules.py [hunk 1]
```diff
@@ -213,10 +213,11 @@ def fulfill_product(self, order, lines):
                 # Collect the EnterpriseCustomer UUID from the coupon, if any.
                 enterprise_customer_uuid = None
                 for discount in order.discounts.all():
-                    enterprise_customer_uuid = discount.voucher.benefit.range.enterprise_customer
-                    if enterprise_customer_uuid is not None:
-                        data['enterprise_course_consent'] = True
-                        break
+                    if getattr(discount, "voucher"):
```
Comment (clintonb) [inline ecommerce/extensions/fulfillment/modules.py]: I believe `discount.voucher` works here.

**Review by clintonb — APPROVED** [review]

File: ecommerce/extensions/fulfillment/modules.py (hunk 1 above)
Comment (mattdrayer) [reply ecommerce/extensions/fulfillment/modules.py]: I actually was going to revise to `hasattr()`.  I figured that the `discount.voucher` in a no-voucher scenario would raise an `AttributeError`?

Comment (clintonb) [reply ecommerce/extensions/fulfillment/modules.py]: `discount.voucher` will either be a `Voucher` or `None. See https://github.com/django-oscar/django-oscar/blob/master/src/oscar/apps/order/abstract_models.py#L1133-L1139.

Comment (mattdrayer) [reply ecommerce/extensions/fulfillment/modules.py]: Ah, nice!  Didn't realize it was a defined property -- TIL 😄 -- okay, will revise, then figure out how to write a test featuring a discount w/ no voucher.

**Comment by mattdrayer** [comment]
@clintonb one more look please and then I'll merge

**Review by clintonb — APPROVED** [review]

**Comment by clintonb** [comment]
@mattdrayer please rebase and merge.

**Pull request merged**